### PR TITLE
NanoAOD: move producers to new event getter methods

### PR DESCRIPTION
**Pull request by swertz**
#### PR description:

Update all NanoAOD producers to `Event::get` and `Event::getHandle`.

Also removed the `packedPFCandidates` input from the `IsFromLostTrackMapProducer` as it wasn't used.

I haven't touched a few plugins because they are apparently not used anywhere and should probably be removed in a future PR (`EGMEnergyVarProducer`, `EnergyRingsTableProducer`, `MuonIDTableProducer`, `NativeArrayTableProducer`); as for `GenWeightsTableProducer`, it is being rewritten in #32167.

#### PR validation:

Ran with `runTheMatrix.py -l 136.8523 --sites ''` and
```
cmsDriver.py nanoMaster --step NANO --datatier NANOAODSIM --eventcontent NANOAODSIM --conditions auto:phase1_2017_realistic --era Run2_2017,run2_nanoAOD_106Xv2 --no_exec --mc --filein "/store/relval/CMSSW_10_6_20/RelValZTT_13/MINIAODSIM/106X_mc2017_realistic_v9_miniv2-v1/10000/C80E4E76-F013-BB40-B619-43713A72C188.root" --fileout nano_mc.root -n 10
```

### CODE REVIEW

**Comment by swertz** [comment]
please test

**Comment by swertz** [comment]
please test

**Comment by swertz** [comment]
+1

Nano integration test results show no difference anywhere as expected: https://gitlab.cern.ch/cms-nanoAOD/nanoAOD-integration/-/issues/194#note_6081029

File: PhysicsTools/NanoAOD/plugins/BJetEnergyRegressionVarProducer.cc [hunk 1]
```diff
@@ -132,16 +132,13 @@ template <typename T>
 void BJetEnergyRegressionVarProducer<T>::produce(edm::StreamID streamID,
```
Comment (perrotta) [inline PhysicsTools/NanoAOD/plugins/BJetEnergyRegressionVarProducer.cc:132]: This file has quite several duplicate header includes: could you please profit and clean them up?

**Review by perrotta — COMMENTED** [review]
In addition the the comments below, there are quite several differences in JETMET DQM, and for the mass of the packedPF candidates in reco. Since this is supppsed to be a "technical PR", are they all understood?

**Comment by swertz** [comment]
I realized several outputs of the `BJetEnergyRegressionVarProducer` were not used anywhere, so I have cleaned it further.

> there are quite several differences in JETMET DQM, and for the mass of the packedPF candidates in reco. Since this is supppsed to be a "technical PR", are they all understood?

Some comparisons are flagged as failed but the histograms look rigorously identical, not sure what to make of it:

image

As for the differences in the jet masses, I also don't know what to say... this PR does not touch that at all.

**Comment by swertz** [comment]
+1

**Comment by perrotta** [comment]
+1
- Also spurious differences have disappeared from the test results

**Pull request merged**